cms/djangoapps/contentstore/views/certificates.py

```diff
@@ -0,0 +1,374 @@
+
+        # Return a new Certificate object instance
+        return certificate
+
+    @staticmethod
+    def get_certificates(course):
+        """
+        Retrieve the certificates list from the provided course
+        """
+        # The top-level course field is 'certificates', which contains various properties,
+        # including the actual 'certificates' list that we're working with in this context
+        return course.certificates.get('certificates', [])
+
+    @staticmethod
+    def remove_certificate(request, store, course, certificate_id):
+        """
+        Remove certificate from the course
+        """
+        for index, cert in enumerate(course.certificates['certificates']):
+            if int(cert['id']) == int(certificate_id):
+                course.certificates['certificates'].pop(index)
+                store.update_item(course, request.user.id)
+                break
+
+    # pylint-disable: unused-variable
+    @staticmethod
+    def remove_signatory(request, store, course, certificate_id, signatory_id):
+        """
+        Remove the specified signatory from the provided course certificate
+        """
+        for cert_index, cert in enumerate(course.certificates['certificates']):  # pylint: disable=unused-variable
+            if int(cert['id']) == int(certificate_id):
+                for sig_index, signatory in enumerate(cert.get('signatories')):  # pylint: disable=unused-variable
+                    if int(signatory_id) == int(signatory['id']):
+                        del cert['signatories'][sig_index]
+                        store.update_item(course, request.user.id)
+                        break
+
+
+class Certificate(object):
+    """
+    The logical representation of an individual course certificate
+    """
+    def __init__(self, course, certificate_data):
+        """
+        Instantiate a Certificate object instance using the provided information.
+        """
+        self.course = course
+        self.__certificate_data = certificate_data
+        self.id = certificate_data['id']  # pylint: disable=invalid-name
+
+    @property
+    def certificate_data(self):
+        """
+        Retrieve the locally-stored certificate data from the Certificate object via a helper method
+        """
+        return self.__certificate_data
+
+    @certificate_data.setter
+    def certificate_data(self, value):
+        """
+        Set certificate name.
+        """
+        self.__certificate_data = value
+
+
+@login_required
+@require_http_methods(("GET", "POST"))
+@ensure_csrf_cookie
+def certificates_list_handler(request, course_key_string):
+    """
+    A RESTful handler for Course Certificates
+
+    GET
+        html: return Certificates list page (Backbone application)
+    POST
+        json: create new Certificate
+    """
+    course_key = CourseKey.from_string(course_key_string)
+    store = modulestore()
+    with store.bulk_operations(course_key):
```

Comment:
I question if the bulk_operations usages are necessary. @cpennington can you offer advise on whey they should be used?